src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,661 @@
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* get values from step_mem */
+  *avg_stage = ((sunrealtype)step_mem->nfe) /
+               ((sunrealtype)ark_mem->nst_attempts);
+
+  return (ARK_SUCCESS);
+}
+
+/*===============================================================
+  Private functions attached to ARKODE
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  lsrkStep_SetDefaults:
+
+  Resets all LSRKStep optional inputs to their default values.
+  Does not change problem-defining function pointers or
+  user_data pointer.
+  ---------------------------------------------------------------*/
+int lsrkStep_SetDefaults(ARKodeMem ark_mem)
+{
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+  long int lenrw, leniw;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* Remove current SUNAdaptController object, and replace with "PI" */
+  if (ark_mem->hadapt_mem->owncontroller)
+  {
+    retval = SUNAdaptController_Space(ark_mem->hadapt_mem->hcontroller, &lenrw,
+                                      &leniw);
+    if (retval == SUN_SUCCESS)
+    {
+      ark_mem->liw -= leniw;
+      ark_mem->lrw -= lenrw;
+    }
+    retval = SUNAdaptController_Destroy(ark_mem->hadapt_mem->hcontroller);
+    ark_mem->hadapt_mem->owncontroller = SUNFALSE;
+    if (retval != SUN_SUCCESS)
+    {
+      arkProcessError(ark_mem, ARK_MEM_FAIL, __LINE__, __func__, __FILE__,
+                      "SUNAdaptController_Destroy failure");
+      return (ARK_MEM_FAIL);
+    }
+  }
+  ark_mem->hadapt_mem->hcontroller = SUNAdaptController_PI(ark_mem->sunctx);
+  if (ark_mem->hadapt_mem->hcontroller == NULL)
+  {
+    arkProcessError(ark_mem, ARK_MEM_FAIL, __LINE__, __func__, __FILE__,
+                    "SUNAdaptControllerPI allocation failure");
+    return (ARK_MEM_FAIL);
+  }
+  ark_mem->hadapt_mem->owncontroller = SUNTRUE;
+  retval = SUNAdaptController_Space(ark_mem->hadapt_mem->hcontroller, &lenrw,
+                                    &leniw);
+  if (retval == SUN_SUCCESS)
+  {
+    ark_mem->liw += leniw;
+    ark_mem->lrw += lenrw;
+  }
+
+  /* Set default values for integrator optional inputs
+     (overwrite some adaptivity params for LSRKStep use) */
+  step_mem->req_stages = 0; /* no stages */
+
+  /* Spectral info */
+  step_mem->lambdaR      = 0;
+  step_mem->lambdaI      = 0;
+  step_mem->sprad        = 0;
+  step_mem->spr_max      = 0;
+  step_mem->spr_min      = 0;
+  step_mem->dom_eig_sfty = 1.01;
+  step_mem->dom_eig_freq = 25;
```

Comment:
Default parameter values usually have a macro defined in the `impl.h` file e.g.,
```
#define DOM_EIG_SAFETY_DEFAULT SUN_RCONST(1.01)
#define DOM_EIG_FREQ_DEFAULT 25
```
These default values need to be noted in the documentation

Reply:
Where do you think the best place to put them in the documentation? @drreynolds @gardner48

Reply:
There are also tables at the start of the "Set" functions section that lists the functions and the default values

Reply:
These values are now in `impl.h`, so users do not know their macro definitions. I have also added their default  values in the documentation with notes right after each set function.